internal/sapcontrol/webservice.go

```diff
@@ -0,0 +1,123 @@
+package sapcontrol
+
+import (
+	"context"
+	"encoding/xml"
+	"fmt"
+	"net"
+	"net/http"
+	"path"
+	"time"
+
+	"github.com/hooklift/gowsdl/soap"
+)
+
+type STATECOLOR string
+
+const (
+	STATECOLORSAPControlGRAY   STATECOLOR = "SAPControl-GRAY"
+	STATECOLORSAPControlGREEN  STATECOLOR = "SAPControl-GREEN"
+	STATECOLORSAPControlYELLOW STATECOLOR = "SAPControl-YELLOW"
+	STATECOLORSAPControlRED    STATECOLOR = "SAPControl-RED"
+)
+
+type Start struct {
+	XMLName  xml.Name `xml:"urn:SAPControl Start"`
+	Runlevel string   `xml:"runlevel,omitempty" json:"runlevel,omitempty"`
+}
+
+type Stop struct {
+	XMLName      xml.Name `xml:"urn:SAPControl Stop"`
+	Softtimeout  int32    `xml:"softtimeout,omitempty" json:"softtimeout,omitempty"`
+	IsSystemStop int32    `xml:"IsSystemStop,omitempty" json:"IsSystemStop,omitempty"`
+}
+
+type GetProcessList struct {
+	XMLName xml.Name `xml:"urn:SAPControl GetProcessList"`
+}
+
+type GetProcessListResponse struct {
+	XMLName   xml.Name     `xml:"urn:SAPControl GetProcessListResponse"`
+	Processes []*OSProcess `xml:"process>item,omitempty" json:"process>item,omitempty"`
+}
+
+type OSProcess struct {
+	Name        string      `xml:"name,omitempty" json:"name,omitempty"`
+	Description string      `xml:"description,omitempty" json:"description,omitempty"`
+	Dispstatus  *STATECOLOR `xml:"dispstatus,omitempty" json:"dispstatus,omitempty"`
+	Textstatus  string      `xml:"textstatus,omitempty" json:"textstatus,omitempty"`
+	Starttime   string      `xml:"starttime,omitempty" json:"starttime,omitempty"`
+	Elapsedtime string      `xml:"elapsedtime,omitempty" json:"elapsedtime,omitempty"`
+	Pid         int32       `xml:"pid,omitempty" json:"pid,omitempty"`
+}
+
+type StartResponse struct {
+}
+
+type StopResponse struct {
+}
+
+type SAPControlConnector interface {
+	/* Triggers an instance start and returns immediately. */
+	StartContext(ctx context.Context, request *Start) (*StartResponse, error)
+	/* Triggers an instance stop and returns immediately. */
+	StopContext(ctx context.Context, request *Stop) (*StopResponse, error)
```

Comment:
`StartContext` and `StopContext` are meant to start and stop the different type of instances that might be running - application or database - on the host, I guess. Or is there other _contexts_ that can be started/stopped?